src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/RoleManagementController.java

```diff
@@ -25,16 +31,17 @@ public RoleManagementController(RoleAssignmentsService roleAssignmentsService) {
     @ApiOperation(
             value = "Assign roles according to the provided specification",
             nickname = "assignRolesJson",
+            notes = "This returns a map of Prison ID to results of the role assignments",
             consumes = MediaType.APPLICATION_JSON_VALUE
     )
     @ApiResponses({
-            @ApiResponse(code = 204, message = "Successful"),
+            @ApiResponse(code = 200, message = "Ok", response = RoleAssignmentStats.class, responseContainer = "List"),
             @ApiResponse(code = 400, message = "Client error")
     })
     @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity assignRolesJson(@RequestBody RoleAssignmentsSpecification specification) {
-        roleAssignmentsService.updateRoleAssignments(specification);
-        return ResponseEntity.noContent().build();
+    public ResponseEntity<List<RoleAssignmentStats>> assignRolesJson(@RequestBody RoleAssignmentsSpecification specification) {
+        val result = roleAssignmentsService.updateRoleAssignments(specification);
```

Comment:
Could be final.

Reply:
added finals everywhere... something like checkstyle would help avoid having to make these kinds of comments.

Reply:
:+1: